homeassistant/components/transmission/coordinator.py

```diff
@@ -0,0 +1,166 @@
+"""Coordinator for transmssion integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import transmission_rpc
+from transmission_rpc.session import SessionStats
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_LIMIT,
+    CONF_ORDER,
+    DEFAULT_LIMIT,
+    DEFAULT_ORDER,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    EVENT_DOWNLOADED_TORRENT,
+    EVENT_REMOVED_TORRENT,
+    EVENT_STARTED_TORRENT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TransmissionDataUpdateCoordinator(DataUpdateCoordinator[SessionStats]):
+    """Transmission dataupdate coordinator class."""
+
+    config_entry: ConfigEntry
+
+    def __init__(
+        self, hass: HomeAssistant, entry: ConfigEntry, api: transmission_rpc.Client
+    ) -> None:
+        """Initialize the Transmission RPC API."""
+        self.config_entry = entry
+        self.api = api
+        self.host = entry.data[CONF_HOST]
+        self._session: transmission_rpc.Session | None = None
+        self._all_torrents: list[transmission_rpc.Torrent] = []
+        self._completed_torrents: list[transmission_rpc.Torrent] = []
+        self._started_torrents: list[transmission_rpc.Torrent] = []
+        self.torrents: list[transmission_rpc.Torrent] = []
+        super().__init__(
+            hass,
+            name=f"{DOMAIN} - {self.host}",
+            logger=_LOGGER,
+            update_interval=timedelta(seconds=self.scan_interval),
+        )
+
+    @property
+    def scan_interval(self) -> float:
+        """Return scan interval."""
+        return self.config_entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
```

Comment:
Make a preliminary PR and remove `scan_interval` from from the config entry options

Reply:
Please refer to this comment (https://github.com/home-assistant/core/pull/98858#issuecomment-1694666182). There is already a PR for this.